core/src/main/java/org/mule/runtime/core/api/exception/FlowExceptionHandler.java

```diff
@@ -50,5 +51,17 @@
       return error(propagateWrappingFatal(throwable));
     }
   }
+
+  /**
+   * Routes the error towards the destination error handler, calling the corresponding callback in case of failure or success.
+   * 
+   * @param error the {@link Exception} to route
+   * @param handledConsumer the callback called in case the error is successfully handled
+   * @param failureConsumer the callback is called in case the error-handling fails
+   */
+  default void routeMessagingError(Exception error, Consumer<CoreEvent> handledConsumer,
```

Comment:
The `handleException` yes, but I think the apply no. Since FlowExceptionHandler is itself a Function, there are some places where it should remain like that for back compatibility.